File: sdks/java/src/main/java/dev/atap/sdk/exception/ATAPRateLimitException.java
package dev.atap.sdk.exception;

import dev.atap.sdk.model.ProblemDetail;

/**
 * Rate limit exceeded (429).
 */
public class ATAPRateLimitException extends ATAPException {

    private final ProblemDetail problem;

    public ATAPRateLimitException(String message, ProblemDetail problem) {
        super(message, 429);
        this.problem = problem;
    }

    public ATAPRateLimitException(String message) {
        this(message, null);
    }

    public ProblemDetail getProblem() {
        return problem;
    }
}

File: sdks/php/src/Exception/ATAPRateLimitException.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Exception;

use Atap\Sdk\Model\ProblemDetail;

/**
 * Rate limit exceeded (429).
 */
class ATAPRateLimitException extends ATAPException
{
    public function __construct(
        string $message = 'Rate limit exceeded',
        public readonly ?ProblemDetail $problem = null,
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, 429, $previous);
    }
}
